sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/tests/Microsoft.OpenTelemetry.Exporter.AzureMonitor.Integration.Tests/TelemetryItemTests.cs

```diff
@@ -137,19 +135,87 @@ private TelemetryItem RunLoggerTest(Action<ILogger<TelemetryItemTests>> testScen
                     .AddProcessor(processor));
             });
 
-            // ACT
             using var serviceProvider = serviceCollection.BuildServiceProvider();
             var logger = serviceProvider.GetRequiredService<ILogger<TelemetryItemTests>>();
+
+            // ACT
             testScenario(logger);
 
             // CLEANUP
             processor.ForceFlush();
-            //Task.Delay(100).Wait(); //TODO: HOW TO REMOVE THIS WAIT?
 
             Assert.True(transmitter.TelemetryItems.Any(), "test project did not capture telemetry");
             return transmitter.TelemetryItems.Single();
         }
 
-        // TODO: INCLUDE ADDITIONAL TESTS VALIDATING ILOGGER + ACTIVITY
+        /// <summary>
+        /// This test is to verify that the correlation Ids from Activity are set on Log Telemetry.
+        /// This test will generate two telemetry items and their Ids are expected to match.
+        /// This test does not inspect any other fields.
+        /// </summary>
+        [Fact]
+        public void VerifyLoggerWithActivity()
+        {
+            // SETUP
+            var ActivitySourceName = "MyCompany.MyProduct.MyLibrary";
```

Comment:
curious - why would we need to have a local activity source instead of a static one that is shared among test cases, is this to allow test cases to run concurrently?

Reply:
I'm hoping to avoid any side effects by containing all objects within the test itself. 
If you think this would be a best practice, I'm happy to change it.

Reply:
The name will cause conflict with other test cases - https://github.com/Azure/azure-sdk-for-net/blob/e7bfbf540ebf93e15a4ba8c01a22df5369536894/sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/tests/Microsoft.OpenTelemetry.Exporter.AzureMonitor.Integration.Tests/TelemetryItemTests.cs#L94

I suggest that we have a clear goal - if we want all the test cases to share the same activity source, or each test case should have its own one. And based on the decision, follow up on what's the correct implementation/abstraction.

Reply:
since the goal is to validate that TraceId/SpanId from LogRecord is correctly exporter, it may be simple to just do 
```
var act = new Activity("foo").Start()
logger.logsomething
act.Stop();
```
And validate that the logrecord has same traceid/span as that of `act`. This should make the test independently validate LogExporting parts.

(If the intention is pure integration test, then okay with current approach).

Reply:
@reyang Done.

@cijothomas Wow that was easy. I added that to the other test. I want this test to be a full integration test.